### PR TITLE
fix(pci.projects.new): post accept contracts before creating

### PR DESCRIPTION
**Pull request by Jisay**

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/new/new.service.js [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
   /* @ngInject */
-  constructor(OvhApiCloud, OvhApiMeAgreements, OvhApiMeVoucherAccount) {
+  constructor($q, OvhApiCloud, OvhApiMeAgreements, OvhApiMeVoucherAccount) {
+    this.$q = $q;
     this.OvhApiCloud = OvhApiCloud;
     this.OvhApiMeAgreements = OvhApiMeAgreements;
     this.OvhApiMeVoucherAccount = OvhApiMeVoucherAccount;
   }
 
+  acceptAgreements(contactList = []) {
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/new/new.service.js]: I think you can use the destructuring assignment and object shorthand such as:

```js
  acceptAgreements(contactList = []) {
    const acceptPromises = map(contactList, ({ id }) => this.OvhApiMeAgreements
      .v6()
      .accept({
        id,
      }, {}).$promise);

     return this.$q.all(acceptPromises);
  }
```

File: packages/manager/modules/pci/src/projects/new/new.routing.js [hunk 2]
```diff
@@ -32,6 +33,17 @@ export default /* @ngInject */ ($stateProvider) => {
       },
       resolve: {
         breadcrumb: () => null,
+        contracts: /* @ngInject */ ($q, newProjectInfo, PciProjectNewService) => {
+          const agreementPromises = map(
+            newProjectInfo.agreements || [],
+            agreementId => PciProjectNewService.getNewProjectAgreementContract(agreementId)
```
Comment (jleveugle) [inline packages/manager/modules/pci/src/projects/new/new.routing.js]:
```suggestion
            id => PciProjectNewService.getNewProjectAgreementContract(agreementId)
```

File: packages/manager/modules/pci/src/projects/new/new.routing.js [hunk 3]
```diff
@@ -32,6 +33,17 @@ export default /* @ngInject */ ($stateProvider) => {
...
+              .then(contract => Object.assign(contract, {
+                id: agreementId,
```
Comment (jleveugle) [inline packages/manager/modules/pci/src/projects/new/new.routing.js]:
```suggestion
                id,
```

File: packages/manager/modules/pci/src/projects/new/new.service.js [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
+    const acceptPromises = map(contactList, contract => this.OvhApiMeAgreements
```
Comment (jleveugle) [inline packages/manager/modules/pci/src/projects/new/new.service.js]:
```suggestion
    const acceptPromises = map(contactList, ({ id }) => this.OvhApiMeAgreements
```

File: packages/manager/modules/pci/src/projects/new/new.service.js [hunk 5]
```diff
@@ -1,11 +1,24 @@
...
+      .v6()
+      .accept({
+        id: contract.id,
```
Comment (jleveugle) [inline packages/manager/modules/pci/src/projects/new/new.service.js]:
```suggestion
        id,
```

**Pull request merged**